File: medicalcheckrest/WMedicalCheck/WMedicalCheck/medicalcheck/src/main/java/ch/hearc/medicalcheck/service/NotificationService.java

package ch.hearc.medicalcheck.service;

import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import ch.hearc.medicalcheck.model.Notification;
import ch.hearc.medicalcheck.repository.NotificationRepository;

/*
* Project   : Medical Check Rest
* Authors   : William Bikuta, Milán Cerviño, Ilyas Boillat, David Oktay
* Date      : 28.01.2022
* Class     : INF3dlm-a
* */

/**
 * bridge between repository and controller
 * each methods is explained in the associate controller
 * check them for more information
 */

@Service
public class NotificationService {
	@Autowired
	private NotificationRepository notificationRepository;

	public List<Notification> getAll() {
		return notificationRepository.findAll();
	}

	public Notification get(Integer id) {
		return notificationRepository.findById(id).get();
	}

	public List<Notification> getAllByIdUser(Integer iduser) {
		return notificationRepository.getAllNotificationByIdUser(iduser);
	}

	public Notification create(Notification notification) {
		return notificationRepository.save(notification);
	}

	public Notification update(Integer idNotification, Notification notification) {
		Optional<Notification> notificationData = notificationRepository.findById(notification.getId());

		if (notificationData.isPresent()) {
			notification.setId(idNotification);
			return notificationRepository.save(notification);
		}
		return null; 
	}

	public void delete(Integer id) {
		notificationRepository.deleteById(id);
	}
}

File: medicalcheck/app/src/main/java/ch/hearc/medicalcheck/services/NotificationService.kt
package ch.hearc.medicalcheck.services

import ch.hearc.medicalcheck.models.Notification
import ch.hearc.medicalcheck.models.User
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.Response
import retrofit2.http.*

/*
* Project   : Medical Check Mobile
* Authors   : William Bikuta, Milán Cerviño, Ilyas Boillat, David Oktay
* Date      : 28.01.2022
* Class     : INF3dlm-a
* */

interface  NotificationService {
    @GET("notifications")
    fun getAll(): Call<List<Notification>>

    @GET("notifications/{id}")
    fun get(@Path("id") id : Int): Call<Notification>

    @GET("notifications/user/{iduser}")
    fun getAllByIdUser(@Path("iduser") iduser : Int): Call<List<Notification>>

    @POST("notifications")
    fun create(@Body notification: Notification): Call<Notification>

    @DELETE("notifications/{id}")
    fun delete(@Path("id") id: Int): Call<ResponseBody>

    @PUT("notifications/{id}")
    fun update(@Path("id") id : Int, @Body notification: Notification): Call<Notification>
}